### ISSUE TITLE
New app Webpack warning

### ISSUE THREAD

**Issue by paganotoni** [gobuffalo/cli#36]
Hello! Thank you so much for your interest in Buffalo. The fact you care enough to be here, right now, helping, makes us very happy.

### Description
Running newly generated app shows webpack 5 warning:

```
(node:22444) [DEP_WEBPACK_DEPRECATION_ARRAY_TO_SET] DeprecationWarning: Compilation.chunks was changed from Array to Set (using Array method 'map' is deprecated)
```

### Steps to Reproduce the Problem

  1. buffalo new app
  2. cd app
  3. buffalo new
  4. buffalo dev

### Expected Behavior
App should start without any warning or error.

### Actual Behavior

App start and shows  a warning:
```
(node:22444) [DEP_WEBPACK_DEPRECATION_ARRAY_TO_SET] DeprecationWarning: Compilation.chunks was changed from Array to Set (using Array method 'map' is deprecated)
```

### Info

Please run `buffalo info` and paste the information below where it says "PASTE_HERE".

<details>

```
-> Go: Checking installation
✓ The `go` executable was found on your system at: /usr/local/go/bin/go

-> Go: Checking minimum version requirements
✓ Your version of Go, 1.17.1, meets the minimum requirements.

-> Go: Checking Package Management
✓ You are using Go Modules (`go`) for package management.

-> Go: Checking PATH
✓ Your PATH contains /Users/paganotoni/go/bin.

-> Node: Checking installation
✓ The `node` executable was found on your system at: /usr/local/bin/node

-> Node: Checking minimum version requirements
✓ Your version of Node, v15.7.0, meets the minimum requirements.

-> NPM: Checking installation
✓ The `npm` executable was found on your system at: /usr/local/bin/npm

-> NPM: Checking minimum version requirements
✓ Your version of NPM, 7.4.3, meets the minimum requirements.

-> Yarn: Checking installation
✓ The `yarnpkg` executable was found on your system at: /Users/paganotoni/.yarn/bin/yarnpkg

-> Yarn: Checking minimum version requirements
✓ Your version of Yarn, 1.22.5, meets the minimum requirements.

-> PostgreSQL: Checking installation
✓ The `postgres` executable was found on your system at: /Applications/Postgres.app/Contents/Versions/latest/bin/postgres

-> PostgreSQL: Checking minimum version requirements
✓ Your version of PostgreSQL, 14.0.0, meets the minimum requirements.

-> MySQL: Checking installation
✘ The `mysql` executable could not be found on your system.
For help setting up your MySQL environment please follow the instructions for you platform at:

https://www.mysql.com/downloads/

-> SQLite3: Checking installation
✓ The `sqlite3` executable was found on your system at: /usr/bin/sqlite3

-> SQLite3: Checking minimum version requirements
✓ Your version of SQLite3, 3.36.0, meets the minimum requirements.

-> Cockroach: Checking installation
✘ The `cockroach` executable could not be found on your system.
For help setting up your Cockroach environment please follow the instructions for you platform at:

https://www.cockroachlabs.com/docs/stable/

-> Buffalo (CLI): Checking installation
✓ The `buffalo` executable was found on your system at: /Users/paganotoni/go/bin/buffalo

-> Buffalo (CLI): Checking minimum version requirements
✓ Your version of Buffalo (CLI), v0.17.5, meets the minimum requirements.

-> Buffalo: Application Details
Pwd         /Users/paganotoni/code/paganotoni/app
Root        /Users/paganotoni/code/paganotoni/app
GoPath      /Users/paganotoni/go
PackagePkg  app
ActionsPkg  app/actions
ModelsPkg   app/models
GriftsPkg   app/grifts
WithModules true
Name        app
Bin         bin/app
VCS         git
WithPop     true
WithSQLite  false
WithDep     false
WithWebpack true
WithNodeJs  true
WithYarn    true
WithDocker  true
WithGrifts  true
AsWeb       true
AsAPI       false
InApp       true
PackageJSON {map[build:webpack --mode production --progress dev:webpack --watch]}

-> Buffalo: config/buffalo-app.toml
name = "app"
bin = "bin/app"
vcs = "git"
with_pop = true
with_sqlite = false
with_dep = false
with_webpack = true
with_nodejs = true
with_yarn = true
with_docker = true
with_grifts = true
as_web = true
as_api = false

-> Buffalo: config/buffalo-plugins.toml
[[plugin]]
  binary = "buffalo-pop"
  go_get = "github.com/gobuffalo/buffalo-pop/v2"

-> Buffalo: go.mod
module app

go 1.17

require (
        github.com/gobuffalo/buffalo v0.17.3
        github.com/gobuffalo/buffalo-pop/v2 v2.3.0
        github.com/gobuffalo/envy v1.9.0
        github.com/gobuffalo/mw-csrf v1.0.0
        github.com/gobuffalo/mw-forcessl v0.0.0-20200131175327-94b2bd771862
        github.com/gobuffalo/mw-i18n v1.1.0
        github.com/gobuffalo/mw-paramlogger v1.0.0
        github.com/gobuffalo/packr/v2 v2.8.1
        github.com/gobuffalo/pop/v5 v5.3.4
        github.com/gobuffalo/suite v2.8.2+incompatible
        github.com/markbates/grift v1.5.0
        github.com/unrolled/secure v1.0.9
)

require (
        github.com/BurntSushi/toml v0.4.1 // indirect
        github.com/Masterminds/semver/v3 v3.1.1 // indirect
        github.com/cockroachdb/cockroach-go v0.0.0-20190925194419-606b3d062051 // indirect
        github.com/davecgh/go-spew v1.1.1 // indirect
        github.com/dustin/go-humanize v1.0.0 // indirect
        github.com/fatih/color v1.12.0 // indirect
        github.com/fatih/structs v1.1.0 // indirect
        github.com/felixge/httpsnoop v1.0.1 // indirect
        github.com/fsnotify/fsnotify v1.4.9 // indirect
        github.com/go-sql-driver/mysql v1.5.0 // indirect
        github.com/gobuffalo/events v1.4.1 // indirect
        github.com/gobuffalo/fizz v1.10.0 // indirect
        github.com/gobuffalo/flect v0.2.3 // indirect
        github.com/gobuffalo/genny v0.6.0 // indirect
        github.com/gobuffalo/github_flavored_markdown v1.1.0 // indirect
        github.com/gobuffalo/helpers v0.6.2 // indirect
        github.com/gobuffalo/httptest v1.5.0 // indirect
        github.com/gobuffalo/logger v1.0.4 // indirect
        github.com/gobuffalo/meta v0.3.0 // indirect
        github.com/gobuffalo/nulls v0.4.0 // indirect
        github.com/gobuffalo/packd v1.0.0 // indirect
        github.com/gobuffalo/plush v3.8.3+incompatible // indirect
        github.com/gobuffalo/plush/v4 v4.1.6 // indirect
        github.com/gobuffalo/pop v4.13.1+incompatible // indirect
        github.com/gobuffalo/tags/v3 v3.1.0 // indirect
        github.com/gobuffalo/validate v2.0.4+incompatible // indirect
        github.com/gobuffalo/validate/v3 v3.1.0 // indirect
        github.com/gofrs/uuid v3.2.0+incompatible // indirect
        github.com/gorilla/handlers v1.5.1 // indirect
        github.com/gorilla/mux v1.8.0 // indirect
        github.com/gorilla/securecookie v1.1.1 // indirect
        github.com/gorilla/sessions v1.2.1 // indirect
        github.com/inconshreveable/mousetrap v1.0.0 // indirect
        github.com/jackc/chunkreader/v2 v2.0.1 // indirect
        github.com/jackc/pgconn v1.8.1 // indirect
        github.com/jackc/pgio v1.0.0 // indirect
        github.com/jackc/pgpassfile v1.0.0 // indirect
        github.com/jackc/pgproto3/v2 v2.0.6 // indirect
        github.com/jackc/pgservicefile v0.0.0-20200714003250-2b9c44734f2b // indirect
        github.com/jackc/pgtype v1.7.0 // indirect
        github.com/jackc/pgx/v4 v4.11.0 // indirect
        github.com/jmoiron/sqlx v1.3.3 // indirect
        github.com/joho/godotenv v1.3.0 // indirect
        github.com/karrick/godirwalk v1.16.1 // indirect
        github.com/kballard/go-shellquote v0.0.0-20180428030007-95032a82bc51 // indirect
        github.com/lib/pq v1.10.1 // indirect
        github.com/luna-duclos/instrumentedsql v1.1.3 // indirect
        github.com/markbates/errx v1.1.0 // indirect
        github.com/markbates/oncer v1.0.0 // indirect
        github.com/markbates/refresh v1.11.1 // indirect
        github.com/markbates/safe v1.0.1 // indirect
        github.com/markbates/sigtx v1.0.0 // indirect
        github.com/mattn/go-colorable v0.1.8 // indirect
        github.com/mattn/go-isatty v0.0.12 // indirect
        github.com/mattn/go-sqlite3 v2.0.3+incompatible // indirect
        github.com/microcosm-cc/bluemonday v1.0.2 // indirect
        github.com/mitchellh/go-homedir v1.1.0 // indirect
        github.com/monoculum/formam v0.0.0-20210523135142-1af3317b7b9b // indirect
        github.com/pelletier/go-toml v1.9.3 // indirect
        github.com/pkg/errors v0.9.1 // indirect
        github.com/pmezard/go-difflib v1.0.0 // indirect
        github.com/rogpeppe/go-internal v1.5.2 // indirect
        github.com/sergi/go-diff v1.1.0 // indirect
        github.com/sirupsen/logrus v1.8.1 // indirect
        github.com/sourcegraph/annotate v0.0.0-20160123013949-f4cad6c6324d // indirect
        github.com/sourcegraph/syntaxhighlight v0.0.0-20170531221838-bd320f5d308e // indirect
        github.com/spf13/cobra v1.2.1 // indirect
        github.com/spf13/pflag v1.0.5 // indirect
        github.com/stretchr/testify v1.7.0 // indirect
        golang.org/x/crypto v0.0.0-20210322153248-0c34fe9e7dc2 // indirect
        golang.org/x/net v0.0.0-20210405180319-a5a99cb37ef4 // indirect
        golang.org/x/sync v0.0.0-20210220032951-036812b2e83c // indirect
        golang.org/x/sys v0.0.0-20210510120138-977fb7262007 // indirect
        golang.org/x/term v0.0.0-20201210144234-2321bbc49cbf // indirect
        golang.org/x/text v0.3.5 // indirect
        gopkg.in/yaml.v2 v2.4.0 // indirect
        gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
)

```

</details>

**Comment by sio4** [comment]
Indeed, I also see the same warning now. I didn't notice it when I upgrade the webpack.

It seems like the dev app started successfully but the webpack watcher printed the warning. However, the webpack worked and the assets were generated successfully.

In detail, I found two deprecation warnings:

```
$ buffalo dev
<...>
(node:757670) [DEP_WEBPACK_TEMPLATE_PATH_PLUGIN_REPLACE_PATH_VARIABLES_HASH] DeprecationWarning: [hash] is now [fullhash] (also consider using [chunkhash] or [contenthash], see documentation for details)
(Use `node --trace-deprecation ...` to show where the warning was created)
<...>
webpack 5.52.1 compiled successfully in 6741 ms
(node:757670) [DEP_WEBPACK_DEPRECATION_ARRAY_TO_SET] DeprecationWarning: Compilation.chunks was changed from Array to Set (using Array method 'map' is deprecated)
```

To check the call stack, ran `NODE_OPTIONS="--trace-deprecation" buffalo dev` instead of `buffalo dev` and got following:

```
(node:763673) [DEP_WEBPACK_TEMPLATE_PATH_PLUGIN_REPLACE_PATH_VARIABLES_HASH] DeprecationWarning: [hash] is now [fullhash] (also consider using [chunkhash] or [contenthash], see documentation for details)
    at /home/sio4/coke/node_modules/webpack/lib/TemplatedPathPlugin.js:90:3
    at /home/sio4/coke/node_modules/webpack/lib/TemplatedPathPlugin.js:295:12
    at String.replace (<anonymous>)
    at replacePathVariables (/home/sio4/coke/node_modules/webpack/lib/TemplatedPathPlugin.js:288:14)
    at Hook.eval [as call] (eval at create (/home/sio4/coke/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:7:16)
    at Compilation.getAssetPathWithInfo (/home/sio4/coke/node_modules/webpack/lib/Compilation.js:3996:40)
    at Compilation.getPathWithInfo (/home/sio4/coke/node_modules/webpack/lib/Compilation.js:3972:15)
    at /home/sio4/coke/node_modules/webpack/lib/Compilation.js:3861:35
    at /home/sio4/coke/node_modules/webpack/lib/Cache.js:91:34
    at Array.<anonymous> (/home/sio4/coke/node_modules/webpack/lib/cache/MemoryCachePlugin.js:45:13)
```

which is related to `tapable`, and...

```
(node:763673) [DEP_WEBPACK_DEPRECATION_ARRAY_TO_SET] DeprecationWarning: Compilation.chunks was changed from Array to Set (using Array method 'map' is deprecated)
    at Set.set.<computed> [as map] (/home/sio4/coke/node_modules/webpack/lib/util/deprecation.js:85:4)
    at /home/sio4/coke/node_modules/webpack-clean-obsolete-chunks/index.js:51:37
    at Array.reduce (<anonymous>)
    at CleanObsoleteChunks._retrieveAllChunks (/home/sio4/coke/node_modules/webpack-clean-obsolete-chunks/index.js:49:23)
    at CleanObsoleteChunks._getObsoleteFiles (/home/sio4/coke/node_modules/webpack-clean-obsolete-chunks/index.js:41:23)
    at CleanObsoleteChunks._removeObsoleteFiles (/home/sio4/coke/node_modules/webpack-clean-obsolete-chunks/index.js:28:28)
    at Hook.eval [as callAsync] (eval at create (/home/sio4/coke/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:20:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/home/sio4/coke/node_modules/tapable/lib/Hook.js:18:14)
    at /home/sio4/coke/node_modules/webpack/lib/Compiler.js:853:27
    at /home/sio4/coke/node_modules/neo-async/async.js:2818:7
```

which is also related to `tapable`.


I found that `tapable` is a project under webpack and the recent version 2.2.1 is already in use by the buffalo app. Since this is a deprecation warning, I think it will not affect the application generated by buffalo now. Anyway, I will check some more about this issue later.

**Comment by sio4** [comment]
Since those two warnings are webpack internal deprecation warnings, I think we can ignore them and can close this issue. It will not affect the results of the application assets, and maybe the future release will fix them.

**Comment by fasmat** [comment]
@paganotoni I think this is resolved with 0.18.0?

**Comment by sio4** [comment]
This is Webpack internal which is some Webpack components still use old things. also does not affect runtime applications but just build-time warnings for deprecated things.

I think we can close this.
